soap/src/main/java/feign/soap/SOAPEncoder.java

```diff
@@ -124,6 +124,9 @@ public void encode(Object object, Type bodyType, RequestTemplate template) {
           Boolean.toString(writeXmlDeclaration));
       soapMessage.setProperty(SOAPMessage.CHARACTER_SET_ENCODING, charsetEncoding.displayName());
       soapMessage.getSOAPBody().addDocument(document);
+
+      modifySOAPMessage(soapMessage);
```

Comment:
can we change this to 
`soapMessage = modifySOAPMessage(soapMessage);`

Assuming that would be help in case someone want to build a `SOAPMessage` from scratch

Reply:
Great idea! I like it.

Regarding the changed headers: I updated them using the maven plugin. Maybe some of them were not required to be updated. Should I revert them?